### PR TITLE
Adding Dynamic Gui for simulator and rewriting for cmake

### PR DESCRIPTION
**Pull request by m-karcz**

### CODE REVIEW

**Comment by van9konst** [comment]
1. `release ` folder do not exists
`cmake ../ProTouchApiSimulator -DCMAKE_BUILD_TYPE=Release`
2. please check for debug symbols in released exe file
`objdump --syms ProTouchApiSimulator.exe`

File: DynamicGui/DynamicGuiController.cpp [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+{
+    pull.unbind(PULL_ADDR);
+    push.disconnect(PUSH_ADDR);
+}
+
+
+QString DynamicGuiController::defaultLeftPanelConfig() const
+{
+    static auto msg = QStringLiteral(
+R"AAAA([
```
Comment (rafalgliniak) [inline DynamicGui/DynamicGuiController.cpp:31]: image

File: DynamicGui/DynamicGuiController.h [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#ifndef DYNAMICGUICONTROLLER_H
```
Comment (rafalgliniak) [inline DynamicGui/DynamicGuiController.h]: Use one type of guards in whole project.

File: CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,52 @@
+project(ProTouchApiSimulator)
```
Comment (rafalgliniak) [inline CMakeLists.txt:1]: Readme has to be changed after dynamic_gui_tester.py is removed

Comment (m-karcz) [reply CMakeLists.txt:1]: adjusted

**Review by rafalgliniak — APPROVED** [review]

**Pull request merged**